File: Lab3/src/utils/CasinoRoyale.java
package utils;

import com.google.gson.Gson;
import model.Account;
import model.Result;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class CasinoRoyale {

    private static final String URI = "http://95.217.177.249/casino/";

    public static Account createAcc() {
        Account account = null;

        Gson gson = new Gson();
        int id = 0;

        while (account == null) {
            try {
                String jsonString = getUrlString(URI + "createacc?id=" + id);
                account = gson.fromJson(jsonString, Account.class);
            } catch (IOException e) {
                id++;
            }
        }

        return account;
    }

    public static Result playLcg(int id, long bed, int number) {
        return play("Lcg", id, bed, number);
    }

    public static Result playMt(int id, long bed, int number) {
        return play("Mt", id, bed, number);
    }

    public static Result playBetterMt(int id, long bed, int number) {
        return play("BetterMt", id, bed, number);
    }

    private static Result play(String mode, int id, long bed, int number) {
        Result result = null;

        Gson gson = new Gson();

        try {
            String jsonString = getUrlString(String.format("%splay%s?id=%s&bet=%s&number=%s", URI, mode, id, bed, number));
            result = gson.fromJson(jsonString, Result.class);
        } catch (IOException e) {
        }

        return result;
    }

    public static String getUrlString(String urlSpec) throws IOException {
        URL url = new URL(urlSpec);

        HttpURLConnection connection = (HttpURLConnection) url.openConnection();

        try {
            BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));

            String inputLine;
            StringBuilder response = new StringBuilder();
            while ((inputLine = in.readLine()) != null) {
                response.append(inputLine);
            }
            in.close();

            return response.toString();

        } catch (IOException e) {
            StringBuilder error = new StringBuilder();
            InputStream errorStream = connection.getErrorStream();

            if (errorStream != null) {
                BufferedReader in = new BufferedReader(new InputStreamReader(errorStream));

                String errorLine;

                while ((errorLine = in.readLine()) != null) {
                    error.append(errorLine);
                }
                in.close();

            } else {
                error.append("null");
            }

            System.out.printf("[%s]: %s - %s - %s%n",
                    connection.getResponseCode(), connection.getResponseMessage(), error, urlSpec);

            throw e;
        }
    }
}

File: Lab3/Lab3/Utils/CasinoRoyale.cs
using Lab3.Model;
using Newtonsoft.Json;
using System;
using System.IO;
using System.Net;

namespace Lab3.Utils
{
    class CasinoRoyale
    {
        private const string URI = "http://95.217.177.249/casino/";

        public static Account CreateAcc()
        {
            Account account = null;

            int id = 0;

            while (account == null)
            {
                string jsonString = Get(URI + "createacc?id=" + id);
                if (jsonString != null)
                {
                    account = JsonConvert.DeserializeObject<Account>(jsonString);
                }
                id++;
            }

            return account;
        }

        public static Result PlayLcg(int id, long bed, int number)
        {
            return Play("Lcg", id, bed, number);
        }

        public static Result PlayMt(int id, long bed, uint number)
        {
            return Play("Mt", id, bed, number);
        }

        public static Result PlayBetterMt(int id, long bed, uint number)
        {
            return Play("BetterMt", id, bed, number);
        }

        private static Result Play(String mode, int id, long bed, long number)
        {
            Result result = null;

            string jsonString = Get($"{URI}play{mode}?id={id}&bet={bed}&number={number}");
            if (jsonString != null)
            {
                result = JsonConvert.DeserializeObject<Result>(jsonString);
            }

            return result;
        }

        public static string Get(string uri)
        {
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
            string responseString = null;

            try
            {
                using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
                using (Stream stream = response.GetResponseStream())
                using (StreamReader reader = new StreamReader(stream))
                {
                    responseString = reader.ReadToEnd();
                }
            }
            catch (WebException ex)
            {
                var response = (HttpWebResponse)ex.Response;
                using (var stream = response.GetResponseStream())
                using (var reader = new StreamReader(stream))
                {
                    string errorResponseString = reader.ReadToEnd();
                    Console.WriteLine($"[{response.StatusCode}]: - {errorResponseString} - {uri}");
                }
            }

            return responseString;
        }
    }
}
